code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -761,9 +827,46 @@ Pass a positive integer as an argument to override a bot's default speed.
 		hack += "[emagged == 2 ? "Software compromised! Unit may exhibit dangerous or erratic behavior." : "Unit operating normally. Release safety lock?"]<BR>"
 		hack += "Harm Prevention Safety System: <A href='?src=\ref[src];operation=hack'>[emagged ? "<span class='bad'>DANGER</span>" : "Engaged"]</A><BR>"
 	else if(!locked) //Humans with access can use this option to hide a bot from the AI's remote control panel and PDA control.
-		hack += "Remote network control radio: <A href='?src=\ref[src];operation=remote'>[remote_disabled ? "Disconnected" : "Connected"]</A><BR><BR>"
+		hack += "Remote network control radio: <A href='?src=\ref[src];operation=remote'>[remote_disabled ? "Disconnected" : "Connected"]</A><BR>"
 	return hack
 
+/mob/living/simple_animal/bot/proc/showpai(mob/user)
+	var/eject = ""
+	if((!locked || issilicon(usr) || IsAdminGhost(usr)))
+		if(paicard || allow_pai)
+			eject += "Personality card status: "
+			if(paicard)
+				if(client)
+					eject += "<A href='?src=\ref[src];operation=ejectpai'>Active</A>"
+				else
+					eject += "<A href='?src=\ref[src];operation=ejectpai'>Inactive</A>"
+			else if (!allow_pai || key)
+				eject += "Unavailable"
+			else
+				eject += "Not inserted"
+			eject += "<BR>"
+		eject += "<BR>"
+	return eject
+
+/mob/living/simple_animal/bot/proc/ejectpai(mob/user = null, announce = 1)
+	if(paicard)
+		if(mind && paicard.pai)
+			mind.transfer_to(paicard.pai)
+		else if (paicard.pai)
+			paicard.pai.key = key
+		else
+			ghostize(0) // The pAI card that just got ejected was dead.
+		key = null
+		paicard.forceMove(loc)

```

Comment:
If the bot is inside a thing, then it should be ejected inside that thing.
For instance, a medibot might be inside a closet, when the AI ejects the pai card.